### ISSUE TITLE
Support for SSL?

### ISSUE THREAD

**Issue by MikesTooLz** [archetyped/simple-lightbox#112]
Wordpress 3.8
website: www.miamidadelostpets.com

I had simple lightbox plugin working on my page but recently I added and SSL certificate to my site.

under Settings > General in wordpress I have changed both wordpress address and site address from http to https

Now simple lightbox is not working on the main page. I'm guessing that this is because its looking for https on the main page but links are going to http. Would be great if it could be updated to work with both.

**Comment by tokar86a** [comment]
How is it going whit this? I have recently switch my site (https://www.myabdllife.com/) over to SSL and now simple lightbox not works any more.

**Comment by EdwardN1** [comment]
Hello - I have just finished developing a website which makes extensive use of SLB and the Video Haddon. We installed the SSL and poof - that was the end of SLB. It works on non SSL but not on SSL - a major blow as this is a very prominent brand website, with hours of work which now has to be redone. Is there a workaround available please?

**Comment by archetyped** [comment]
@EdwardN1 please provide a link to a page that exhibits the issue you are experiencing and I would be glad to take a closer look.

**Comment by EdwardN1** [comment]
Thank you for the quick response, at this point the site is in development, close to release, so I am not allow to link to it in a public forum unfortunately - I written something based on magnetic popup as a stop gap, as the client has become very time sensitive. I will try had replicate the issue elsewhere, as a demo at some point if I get  a spare moment!

**Comment by tokar86a** [comment]
How is it going whit this?

**Comment by Keelan** [comment]
I guess this still isn't resolved?

https://relcontrols.ca/products/controllers/proweb-series/41-mach-prowebcom

**Comment by archetyped** [comment]
Hi @Keelan, it appears that cURL is not properly configured/enabled for your site.  Please see SLB's minimum requirements for [more details on cURL support](https://github.com/archetyped/simple-lightbox/wiki/Requirements#curl-support).

Also, your theme appears to have some JavaScript code that is causing an error.  This error is stopping all other JS code (including SLB's) from running.  This will also need to be resolved before SLB can function normally.

**Comment by Keelan** [comment]
Yes after i submitted this n I discovered it seems my theme update has
disrupted a bunch of plugins.

The curl thing is odd though because this did work before.

I'll look into it and report back. Thanks.

**Comment by Artuvazro** [comment]
I'm having the same issue. Can't make it work with HTTPS.
Hereis my cURL info: http://i.imgur.com/6V8Ka40.png

**Comment by EoleDevel** [comment]
Hello,

I think the problem here is not related to Simple Lightbox plugin.

Using SSL/TLS requires a certificate installed on the web server. However, for some performance reasons, some host providers may choose to install the certificate on a load balancer. This affect cURL which cannot not perform local requests using HTTPS since there is no certificate on the server itself.

@Artuvazro: I just looked at your provider and you use OVH (me too !) which is affected by this problem. However, OVH uses a port forwarding on there Web servers for SLL/TLS. So you can perform a local request using http://localizacomopuedas.es:443/?p=770 (use http and specify the port 443). Consequently, I think you need to rewrite all https URL to HTPP with 443 specified (maybe with htaccess or a custom WordPress plugin).

**Comment by Artuvazro** [comment]
@EoleDevel Yes, I could do that, but if all the content is not in https then you don't get the green lock, am I right?

**Comment by EoleDevel** [comment]
I'm talking about local requests (web server to web server). Clients are not aware of this stuff so it doesn't affect the green lock.

**Comment by cafsantos** [comment]
I had same problem.

Using SSL/TLS requires a certificate installed on the web server. If you certificate is not properly installed you could simply add this line in your functions.php

add_filter( 'https_ssl_verify', '__return_false' );

This potencial problem could be avoided if plugin use file_get_contents instead to  wp_safe_remote_get to read template file (simple-lightbox/includes/class.base_object.php)

**Comment by bobbingwide** [comment]
> This potencial problem could be avoided if plugin use file_get_contents instead to wp_safe_remote_get to read template file (simple-lightbox/includes/class.base_object.php)

Alternatively, change line 254 of `includes/class.base_object.php` to
```  
	$get = wp_safe_remote_get($ret, ['sslverify' => false]);
```

I found that I needed to do this in my local installation ( Windows server) where I have SSL enabled,
but have never managed to correctly configure the SSL certificate to support local requests using `https:`.
